Alerting: Add CustomDetails field in PagerDuty contact point (#64860)
grafana/grafana: docs/sources/alerting/manage-notifications/manage-contact-points/configure-notifiers.md

---

> description: Notifier configuration guide

Reviewer: ```suggestion
description: Configure contact points
```

draft: true
keywords:
  - Grafana
  - alerting
  - guide
  - notifications
  - notifiers
  - contact points

> title: Configure notifiers

Reviewer: ```suggestion
title: Configure contact points
```

weight: 100
---

# Configure notifiers

> Configuring notifiers in Grafana allows you to receive notifications for your alert rules in your preferred communication channels. Each notifier has its own configuration options and setup process, but in general, the process involves providing an API key or webhook URL.

Reviewer: ```suggestion
Configure contact points in Grafana to select your preferred communication channel for receiving notifications when your alert rules are fire. Each contact point has its own configuration options and setup process. In most cases, this involves providing an API key or a Webhook URL.
```

> Once configured, you can use notifiers as part of your contact points to receive notifications whenever your alert changes its state. In this section, we'll cover the basic steps to configure your notifiers, so you can start receiving real-time alerts and stay on top of your monitoring data.

Reviewer: Is a notifier different to a contact point? I thought contact points were notifiers? Can you configure your notifiers to receive alert rules in different states - or only when firing?

Reviewer: In which case, I'd write "Once you've configured your contact points, you'll receive  notifications whenever your alert rule changes state.

Author: > Is a notifier different to a contact point? I thought contact points were notifiers?

I agree it's confusing! A contact point is basically a set of notifiers used together. For example, we have [docs](https://grafana.com/docs/grafana/latest/alerting/manage-notifications/webhook-notifier/) for configuring the Webhook notifier, which we can use in a contact point alongside other notifiers.

In upstream Alertmanager terms:
- Contact point -> Receiver.
- Notifier -> Integration.

I just checked and realized that we use the word "integrations" A LOT to refer to notifiers in our docs and in Grafana. It would be great if the naming was more consistent, so I think I'll change this to use that term.

Author: > Can you configure your notifiers to receive alert rules in different states - or only when firing?

You can receive resolved alerts too, this is an opt-out.
You can also receive _NoData_ and _Error_ notifications, which are not the same as _Firing_.

| Name                    | Type                      |
| ----------------------- | ------------------------- |
| DingDing                | `dingding`                |
| Discord                 | `discord`                 |
| Email                   | `email`                   |
| Google Hangouts Chat    | `googlechat`              |
| Hipchat                 | `hipchat`                 |
| Kafka                   | `kafka`                   |
| Line                    | `line`                    |
| Microsoft Teams         | `teams`                   |
| Opsgenie                | `opsgenie`                |
| [Pagerduty](#pagerduty) | `pagerduty`               |
| Prometheus Alertmanager | `prometheus-alertmanager` |
| Pushover                | `pushover`                |
| Sensu                   | `sensu`                   |
| Sensu Go                | `sensugo`                 |
| Slack                   | `slack`                   |
| Telegram                | `telegram`                |
| Threema                 | `threema`                 |
| VictorOps               | `victorops`               |
| Webhook                 | `webhook`                 |

### PagerDuty

> To set up PagerDuty, all you have to do is to provide an integration key.

Reviewer: ```suggestion
To set up PagerDuty, provide an integration key.
```

| Setting         | Description                                            |
| --------------- | ------------------------------------------------------ |
| Integration Key | Integration key for PagerDuty                          |
| Severity        | Level for dynamic notifications, default is `critical` |
| Custom Details  | Additional details about the event                     |

> The `CustomDetails` field is an object containing arbitrary key-value pairs. The user-defined details get merged with the ones we use by default.

Reviewer: ```suggestion
The `CustomDetails` field is an object containing arbitrary key-value pairs. The user-defined details are merged with the ones we use by default.
```

Reviewer: Hi @santihernandezc are we ready to go with this one now? i think the only change pending is contact points / contact point types instead of receivers. thanks!

Author: @brendamuir I was on call last week (and PTO yesterday), so I'm resuming work today.

Regarding terminology, I don't recall ever calling it a "contact point type". We almost always refer to it using the word "Integration", and on the _Contact Points_ page we changed the input label to use the word "Integration" ([link to changelog](https://github.com/grafana/grafana/blob/dd04757fc9e6e6cfdcd2772ac78f8a6bc2e83013/CHANGELOG.md?plain=1#L122)).

![Screen Shot 2023-03-28 at 10 36 02](https://user-images.githubusercontent.com/41638679/228254876-7c84deec-a2c6-4b54-9d01-c3f4b3a8e1b8.png)

The term "contact point type" can be misleading. A contact point is a set of integrations:
- Contact point 1
    - Integration a
    - Integration b
    - Integration c
   
Saying "contact point type" would falsely imply that an integration is at the same level as a contact point. To me, it doesn't make sense to have a Contact Point that is made up of Contact Point Types, which sound like attributes of the Contact Point rather than something completely different that is encapsulated in that contact point.
- Contact point 1
    - Contact point type a
    - Contact point type b
    - Contact point type c

I'll leave a message on Slack to get other opinions on this.
